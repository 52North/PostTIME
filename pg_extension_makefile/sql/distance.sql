CREATE TABLE test4posttime_distance
(
  id serial NOT NULL,
  posttime_1 posttime,
  posttime_2 posttime,
  CONSTRAINT test4posttime_distance_pkey PRIMARY KEY (id)
)
WITH (
  OIDS=FALSE
);

INSERT INTO test4posttime_distance (posttime_1, posttime_2)
VALUES 
('2010-4-12T14:23:42','2010-4-12T14:23:42'),
('2010-5-12T14:23:42','2010-4-12T14:23:42'),
('2010-3-12T14:23:42','2010-4-12T14:23:42'),
('2010-4-12T14:23:42/2010-4-13T14:23:42','2010-5-12T14:23:42'), 
('2010-4-12T14:23:42/2010-4-13T14:23:42','2010-4-13T14:23:42'), 
('2010-4-12T14:23:42/2010-4-13T14:23:42','2010-4-12T17:23:42'), 
('2010-4-12T14:23:42/2010-4-13T14:23:42','2010-4-12T14:23:42'), 
('2010-4-12T14:23:42/2010-4-13T14:23:42','2009-5-12T14:23:42'), 
('1996-5-12T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42'), 
('1997-4-12T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42'), 
('2005-5-12T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42'), 
('2010-4-13T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42'), 
('2012-5-12T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42'), 
('1994-4-12T14:23:42/1996-4-13T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42'), 
('1994-4-12T14:23:42/1997-4-12T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42'), 
('1994-4-12T14:23:42/2005-4-13T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42'), 
('1997-4-12T14:23:42/1999-4-13T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42'), 
('1997-4-12T14:23:42/2012-4-13T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42'), 
('1999-4-12T14:23:42/2001-4-13T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42'), 
('1994-4-12T14:23:42/2012-4-13T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42'), 
('1997-4-12T14:23:42/2010-4-13T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42'), 
('1998-4-12T14:23:42/2012-4-13T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42'), 
('1999-4-12T14:23:42/2010-4-13T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42'), 
('1994-4-12T14:23:42/2010-4-13T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42'), 
('2010-4-13T14:23:42/2012-4-13T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42'), 
('2012-4-13T14:23:42/5454-4-13T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42'),
('2012-9-3T16:33:22.212','2012-9-17T12'),
('2009-7-1/2009-10-1T12','2009-10-1/2012-10-1'),
('TCS0012456157.0157','TCS0012456156.09/2456157.07553'),
('TCS0012456157.07553','TCS0012456156.09'),
('2012-5-14T19:27:22.2','2012-7-14T19:27:22.2'),
('2012-8-14T19:27:22.2','2012-7-14T19:27:22.2'),
('2012-7-14T19:27:22.2','2012-7-14T20:27:22.2'),
('TCS0012111222.345678','TCS0012111223.345678'),
('2010-4-12T14:23:42,2010-4-12T14:23:42','2012'),
('2010-5-12T14:23:42','2010-4-12T14:23:42,2011,2013,2015'),
('2010-3-12T14:23:42/2013,2017/2019','2010-4-12T14:23:42');

SELECT tm_distance(posttime_1, posttime_2) FROM test4posttime_distance;

SELECT tm_distance_dec_day(posttime_1, posttime_2) FROM test4posttime_distance;

SELECT tm_duration(posttime_1) FROM test4posttime_distance;

SELECT tm_duration_dec_day(posttime_1) FROM test4posttime_distance;

