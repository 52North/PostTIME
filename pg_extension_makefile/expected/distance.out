CREATE TABLE test4posttime_distance
(
  posttime_1 posttime,
  posttime_2 posttime,
  dist_dec_day float,
  dist_dec_day_single float
);
INSERT INTO test4posttime_distance (posttime_1, posttime_2, dist_dec_day, dist_dec_day_single)
VALUES 
('2010-4-12T14:23:42','2010-4-12T14:23:42',0,0),
('2010-5-12T14:23:42','2010-4-12T14:23:42',30,0),
('2010-3-12T14:23:42','2010-4-12T14:23:42',31,0),
('2010-4-12T14:23:42/2010-4-13T14:23:42','2010-5-12T14:23:42',29,1), 
('2010-4-12T14:23:42/2010-4-13T14:23:42','2010-4-13T14:23:42',0,1), 
('2010-4-12T14:23:42/2010-4-13T14:23:42','2010-4-12T17:23:42',0,1), 
('2010-4-12T14:23:42/2010-4-13T14:23:42','2010-4-12T14:23:42',0,1), 
('2010-4-12T14:23:42/2010-4-13T14:23:42','2009-5-12T14:23:42',335,1), 
('1996-5-12T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42',335,0), 
('1997-4-12T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42',0,0), 
('2005-5-12T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42',0,0), 
('2010-4-13T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42',4749,0), 
('2012-5-12T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42',760,0), 
('1994-4-12T14:23:42/1996-4-13T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42',364,732), 
('1994-4-12T14:23:42/1997-4-12T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42',0,1096), 
('1994-4-12T14:23:42/2005-4-13T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42',0,4019), 
('1997-4-12T14:23:42/1999-4-13T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42',0,731), 
('1997-4-12T14:23:42/2012-4-13T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42',0,5480), 
('1999-4-12T14:23:42/2001-4-13T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42',0,732), 
('1994-4-12T14:23:42/2012-4-13T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42',0,6576), 
('1997-4-12T14:23:42/2010-4-13T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42',0,4749), 
('1998-4-12T14:23:42/2012-4-13T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42',365,5115), 
('1999-4-12T14:23:42/2010-4-13T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42',730,4019), 
('1994-4-12T14:23:42/2010-4-13T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42',0,5845), 
('2010-4-13T14:23:42/2012-4-13T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42',0,731), 
('2012-4-13T14:23:42/5454-4-13T14:23:42','1997-4-12T14:23:42/2010-4-13T14:23:42',731,1257164),
('2012-9-3T16:33:22.212','2012-9-17T12',13.8101595833333,0),
('2009-7-1/2009-10-1T12','2009-10-1/2012-10-1',0,92.5),
('TCS0012456157.0157','TCS0012456156.09/2456157.07553',0,0),
('TCS0012456157.07553','TCS0012456156.09',0.98553,0),
('2012-5-14T19:27:22.2','2012-7-14T19:27:22.2',61,0),
('2012-8-14T19:27:22.2','2012-7-14T19:27:22.2',31,0),
('2012-7-14T19:27:22.2','2012-7-14T20:27:22.2',0.0416666666666667,0),
('TCS0012111222.345678','TCS0012111223.345678',1,0),
('2010-4-12T14:23:42,2010-4-12T14:23:42','2012',-1,-1),
('2010-5-12T14:23:42','2010-4-12T14:23:42,2011,2013,2015',-1,0),
('2010-3-12T14:23:42/2013,2017/2019','2010-4-12T14:23:42',-1,-1);
SELECT tm_distance(posttime_1, posttime_2) FROM test4posttime_distance;
    tm_distance     
--------------------
 P0
 P1M
 P1M
 P29D
 P0
 P0
 P0
 P11M
 P11M
 P0
 P0
 P13Y1D
 P2Y29D
 P11M29D
 P0
 P0
 P0
 P0
 P0
 P0
 P0
 P1Y
 P2Y
 P0
 P
 P2Y
 P13DT19H26M37.788S
 P0
 P0
 PT23H39M9.792S
 P2M
 P1M
 PT1H
 P1D
 NAP
 NAP
 NAP
(37 rows)

SELECT (tm_distance_dec_day(posttime_1, posttime_2)-dist_dec_day)<0.000001 FROM test4posttime_distance;
 ?column? 
----------
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
(37 rows)

SELECT tm_duration(posttime_1) FROM test4posttime_distance;
 tm_duration 
-------------
 P0
 P0
 P0
 P1D
 P1D
 P1D
 P1D
 P1D
 P0
 P0
 P0
 P0
 P0
 P2Y1D
 P3Y
 P11Y1D
 P2Y1D
 P15Y1D
 P2Y1D
 P18Y1D
 P13Y1D
 P14Y1D
 P11Y1D
 P16Y1D
 P2Y
 P3442Y
 P0
 P3MT12H
 P0
 P0
 P0
 P0
 P0
 P0
 NAP
 P0
 NAP
(37 rows)

SELECT (tm_duration_dec_day(posttime_1)-dist_dec_day_single)<0.000001 FROM test4posttime_distance;
 ?column? 
----------
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
 t
(37 rows)

